### DISCUSSION TITLE
CLOSED[Umbrella] remove SitePage nodes

### DISCUSSION THREAD

**Discussion by KyleAMathews** [gatsbyjs/gatsby#29233] [Ideas / Feature Requests]
We're deprecating creating SitePage nodes and will remove in v3. The reason is that we already create objects for pages so also creating SitePage nodes is redundant work and gets costly as a site's page count grows.

You can still query `sitePage` and `allSitePage` in the graphql schema so plugins requesting that data should still work as before. **NOTE** the filtering/sorting operators that are normally on Gatsby's graphql types has not yet been added to these. If you need one of the operators added to use this, let us know in a comments so we can fast track adding it.

PR implemented in https://github.com/gatsbyjs/gatsby/pull/29201

## How to test

Add the `NO_PAGE_NODES ` flag to your `gatsby-config.js`:

```js
module.exports = {
  flags: {
    NO_PAGE_NODES: true
  },
  plugins: [...]
}
```

**Comment by kimbaudi** [comment]
Does this mean we no longer have access to `SitePage` nodes in `gatsby-node.js`? Because I can still access `SitePage` nodes using `getNodes` method.

Because I added the `NO_PAGE_NODES` flag in `gatsby-config.js` and I think SitePage nodes are still being created. This is my `onPostBootstrap` in `gatsby-node.js` and it still logs the page paths from the SitePage nodes. I really don't have any other use for `SitePage` nodes besides checking pages paths created `onPostBootstrap`, but still surprised that I have access to `SitePage` nodes.

```js
exports.onPostBootstrap = ({ getNodes, reporter }) => {
  reporter.info(`onPostBootstrap`)
  const paths = getNodes()
    .filter((n) => n.internal.type === `SitePage`)
    .map((n) => n.path)
  console.log(paths)
}
```

**Reply by KyleAMathews** [reply]
Oh sorry, just saw this. This still isn't published. Working on getting a revised method out for this

**Comment by kimbaudi** [comment]
> We're deprecating creating SitePage nodes and will remove in v3.

Does this mean `createPage` and `deletePage` methods are deprecated for `onCreatePage` Gatsby Node API? 

```js
exports.onCreatePage = ({ page, actions: { createPage, deletePage } }) => {
  const path = stripTrailingSlash(page.path)
  deletePage(page)
  createPage({
    ...page,
    path,
  })
}
```

I use this not only to remove trailing slashes, but also to add page context to create pages. I hope this is not going to be deprecated.
